### PR TITLE
Fixes and upgrades to the Asterisk binding

### PR DESCRIPTION
**Pull request by MrArancio**
Updated the asterisk-java library to the latest trunk, now the binding works with Asterisk 11
Added some more configuration options:
- retained compatibility with previous configuration
- turn on/off a switch item on specific call (filter by caller and/or callee)
- turn on a switch item on dtmf digit event (filter by caller, callee and digit)

### CODE REVIEW

File: bundles/binding/org.openhab.binding.asterisk/README.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Switch light (lights) {asterisk="digit:*:101:1"}
+
+3) Switch on a light whenever a '1' digit is sent from '215' during an active call
+
+Switch light (lights) {asterisk="digit:215:*:1"}
+
+4) Switch on a light whenever a '1' digit is sent during an active call
+
+Switch light (lights) {asterisk="digit:*:*:1"}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.asterisk/README.txt]: please update the wiki page accordingly

File: bundles/binding/org.openhab.binding.asterisk/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -27,4 +27,4 @@ Bundle-ActivationPolicy: lazy
 Export-Package: org.openhab.binding.asterisk
 Service-Component: OSGI-INF/genericbindingprovider.xml, OSGI-INF/binding.xml
 Bundle-ClassPath: .,
- lib/asterisk-java-1.0.0-20090923.194527-626.jar
+ lib/asterisk-java.jar
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.asterisk/META-INF/MANIFEST.MF]: add the version number to asterisk lib

**Comment by teichsta** [comment]
@MrArancio thanks for this contribution. Please find my comments inline. Please finally squash your commits into one after incorporating my feedback. Thanks, Thomas E.-E.

**Comment by teichsta** [comment]
any update @MrArancio ?

**Comment by teichsta** [comment]
how are we going to proceed @MrArancio? Any chance to incorporate my feedback?

**Comment by 9037568** [comment]
Is @MrArancio still around and able to finish this PR?

**Comment by 9037568** [comment]
@watou 

John, what's the protocol for stale PRs like this one? Can I finish up the needed changes myself?

**Comment by watou** [comment]
Sorry I missed your question Chris!  Great you got this one.

**Comment by 9037568** [comment]
No worries, John. :)

**Comment by 9037568** [comment]
Thanks, @MrArancio for your good work on this.

**Pull request merged**